src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -754,6 +756,105 @@ void ThermostatAttrAccess::OnFabricRemoved(const FabricTable & fabricTable, Fabr
     }
 }
 
+void EmitSetpointEvent(const ConcreteAttributePath & attributePath, SystemModeEnum systemMode,
+                       Optional<BitMask<OccupancyBitmap>> occupancy, SetpointGetter getter)
+{
+    int16_t setpoint;
+    auto status = getter(attributePath.mEndpointId, &setpoint);
+    if (status != Status::Success)
+    {
+        ChipLogError(Zcl, "EmitSetpointChangeEvent failed to queue event: could not get set point");
+        return;
+    }
+    EmitSetpointChangeEvent(attributePath.mEndpointId, systemMode, occupancy, chip::Optional<int16_t>(), setpoint);
+}
+
+void EmitEvents(const ConcreteAttributePath & attributePath)
+{
+    switch (attributePath.mAttributeId)
+    {
+    case SystemMode::Id: {
+        SystemModeEnum systemMode = SystemModeEnum::kOff;
+        if (SystemMode::Get(attributePath.mEndpointId, &systemMode) != Status::Success)
+        {
+            ChipLogError(Zcl, "Failed to queue SystemModeChange event: could not get system mode");
+        }
+        else
+        {
+            EmitSystemModeChangeEvent(attributePath.mEndpointId, chip::Optional<SystemModeEnum>(), systemMode);
+        }
+        break;
+    }
+    case OccupiedHeatingSetpoint::Id:
+        EmitSetpointEvent(attributePath, SystemModeEnum::kHeat,
+                          chip::Optional<chip::BitMask<OccupancyBitmap>>(OccupancyBitmap::kOccupied), OccupiedHeatingSetpoint::Get);
```

Comment:
```suggestion
                          MakeOptional(OccupancyBitmap::kOccupied), OccupiedHeatingSetpoint::Get);
```
if that works.  If not, at least remove the pointless `chip::` prefixes?